### ISSUE TITLE
Feature Request: Support Pod Failure Policy for Jobs / Cronjobs

### ISSUE THREAD

**Issue by snorlaX-sleeps** [hashicorp/terraform-provider-kubernetes#2374]
### Description

Add [Pod Failure Policy](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-failure-policy) configuration to the Kubernetes Job Spec.
As this is only GA in Kubernetes v1.29 I can understand why it may not already be available.

This feature allows handling different kinds of error states and if it is counted towards the `backoffLimit`.

### Potential Terraform Configuration

Using the example from the linked documentation:
```hcl
pod_failure_policy {
  rule {
    action = "FailJob"
    on_exit_codes = {
      container_name = "main"
      operator       = "In"
      values         = [42]
    }
  }
  rule {
    action = "Ignore"
    on_pod_conditions = {
      type = "DisruptionTarget"
    }
  }
}
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by BBBmau** [comment]
Support for `PodFailurePolicy` is only found in the job schema and not in cronjob schema. The linked PR will add this.

**Comment by snorlaX-sleeps** [comment]
@BBBmau - as a cronjob contains a job template, would it be included there? (are all job features not available to a cronjob?)
Thanks for making a PR

**Comment by BBBmau** [comment]
@snorlaX-sleeps Yes since `cronjob` uses job template it would already be included since it uses the same `jobSpec` found in the job resource. Thanks for bringing this up, because of this cronjob tests will need to include the checking of `job_template` to include `PodFailurePolicy`.

Latest commit includes that.
